ivy.el

```diff
@@ -610,6 +619,26 @@ candidate, not the prompt."
            (insert ivy-text)
            (ivy--exhibit)))))
 
+(defvar ivy-mouse-action-tooltip
+  "Exit the minibuffer with the selected candidate."
+  "The doc visible in the tooltip for mouse-1 binding in the minibuffer")
+
+(defun ivy-mouse-done (event)
+  "Exit the minibuffer with the selected candidate."
+  (interactive "@e")
+  (when event
+    (let* ((line-number-at-point
+            (max 2
+                 (line-number-at-pos (posn-point (event-start event)))))
+
+           (line-number-candidate ;; convert to 0 based index
+            (- line-number-at-point 2))
+           (offset
+            (- line-number-candidate
+               ivy--window-index)))
+      (ivy-next-line-and-call offset))
+    (ivy-alt-done)))
```

Comment:
Maybe it's just me, but I find it clearer to write

```el
(defun ivy-mouse-done (event)
  "Exit the minibuffer with the selected candidate."
  (interactive "@e")
  (when event
    (ivy-next-line-and-call
     (- (max 0 (- (line-number-at-pos (posn-point (event-start event)))
                  2))
        ivy--window-index))
    (ivy-alt-done)))
```

Out of curiosity (as I am not familiar with the code or its purpose), why is up to 2 subtracted from the line number?

Reply:
As it is no so clear why the 2 is subtracted, I preferred to give meaningful names to the values.

`line-number-at-point` is the line number in the minibuffer window; line 1 is the prompt, line 2 the first candidate... as the click event designates a candidate, there is `(max 2 ...)`.

`line-number-candidate` is the corresponds to the index of the selected candidate in the list off displayed candidates (which is indexed from 0).

We substract one because of the prompt, and one to convert to 0 based index.

Reply:
Thanks, that makes sense. I'm not against naming intermediate results, I just find it clearer to subtract 2 and then ensure the result is a valid index, rather than doing `(max 2 N)` followed by a subtraction of 2.